memor/prompt.py

```diff
@@ -287,30 +286,49 @@ def from_json(self, json_object: Union[str, Dict[str, Any]]) -> None:
             role = Role(loaded_obj["role"])
             template = PresetPromptTemplate.DEFAULT.value
             if "template" in loaded_obj:
-                template_obj = PromptTemplate()
-                template_obj.from_json(loaded_obj["template"])
-                template = template_obj
+                template = PromptTemplate()
+                template.from_json(loaded_obj["template"])
             memor_version = loaded_obj["memor_version"]
             date_created = datetime.datetime.strptime(loaded_obj["date_created"], DATE_TIME_FORMAT)
             date_modified = datetime.datetime.strptime(loaded_obj["date_modified"], DATE_TIME_FORMAT)
             selected_response_index = loaded_obj["selected_response_index"]
         except Exception:
             raise MemorValidationError(INVALID_PROMPT_STRUCTURE_MESSAGE)
-        if len(responses) > 0:
-            self._selected_response = responses[selected_response_index]
-            self._selected_response_index = selected_response_index
-        else:
-            self._selected_response = None
-            self._selected_response_index = 0
-        self._message = message
-        self._tokens = tokens
-        self._id = _id
-        self._responses = responses
-        self._role = role
-        self._template = template
-        self._memor_version = memor_version
-        self._date_created = date_created
-        self._date_modified = date_modified
+        _validate_string(message, "message")
```

Comment:
I reviewed this PR and have some suggestions for refactoring that we can apply in future PRs. However, I believe we should validate `selected_response_index` like we do for other parameters such as `message`, `tokens`, and others.

Reply:
fixed in 675b455